tst/providers/MeetingProvider/index.test.tsx

```diff
@@ -0,0 +1,78 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import {LogLevel, EventName, EventAttributes, DefaultEventController, NoOpDebugLogger, MeetingSessionConfiguration, MeetingSessionCredentials, MeetingSessionURLs} from "amazon-chime-sdk-js";
+import React from 'react';
+import {renderHook} from '@testing-library/react-hooks'
+import MeetingManager from '../../../src/providers/MeetingProvider/MeetingManager';
+import {MeetingContext, useMeetingManager} from '../../../src/providers/MeetingProvider';
+
+
+describe('Meeting Provider', () => {
+
+  it('events are received correctly', async (done) => {
+    // Mock the user agent to ensure the MeetingManager event subscription is 
+    // set up. Otherwise, the amazon-chime-sdk-js will not detect a valid
+    // browser to use.
+    const userAgentGet = jest.spyOn(navigator, 'userAgent', 'get');
+    userAgentGet.mockReturnValue('Chrome/96.0');
+    
+    // Event details
+    const eventName = 'audioInputFailed';
+    const audioInputErrorMessage = 'Something went wrong';
+
+    // Setup meeting manager and eventcontroller
+    let meetingManager = new MeetingManager({ logLevel: LogLevel.OFF});
+    let logger = new NoOpDebugLogger();
+    let emptyConfiguration = new MeetingSessionConfiguration();
+    emptyConfiguration.meetingId = '';
+    emptyConfiguration.externalMeetingId = '';
+    emptyConfiguration.credentials = new MeetingSessionCredentials();
+    emptyConfiguration.credentials.attendeeId = '';
+    emptyConfiguration.credentials.joinToken = '';
+    emptyConfiguration.urls = new MeetingSessionURLs();
+    emptyConfiguration.urls.turnControlURL = '';
+    emptyConfiguration.urls.audioHostURL = '';
+    emptyConfiguration.urls.signalingURL = 'wss://localhost/';
+    meetingManager.eventController = new DefaultEventController(emptyConfiguration, logger);
+    await meetingManager.initializeMeetingSession(emptyConfiguration);
```

Comment:
We should not be using `initializeMeetingSession` as we want to make it private, in fact Kunal's changes are removing this method. Can you use `join` instead?

Reply:
Updated